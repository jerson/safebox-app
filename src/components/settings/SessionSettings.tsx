import React, { useCallback, useState } from 'react';
import {
  StyleSheet,
  ViewStyle,
  View,
  StyleProp,
  Alert,
  TextStyle
} from 'react-native';
import Colors from '../../modules/constants/Colors';
import SplitText from '../ui/SplitText';
import Button from '../ui/Button';
import Session from '../../services/Session';
import { useNavigation } from 'react-navigation-hooks';
import Text from '../ui/Text';
import Log from '../../modules/log/Log';
import Client from '../../services/Client';

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.grey1
  } as ViewStyle,
  content: {
    marginTop: 10,
    flexDirection: 'row',
    alignItems: 'center'
  } as ViewStyle,
  info: {
    flex: 1,
    paddingRight: 10
  } as ViewStyle,
  description: {
    fontSize: 12,
    color: Colors.grey5
  } as TextStyle,
  button: {
    minWidth: 120
  } as ViewStyle
});

export interface SessionSettingsProps {
  style?: StyleProp<ViewStyle>;
}

const TAG = '[SessionSettings]';
function SessionSettings({ style }: SessionSettingsProps) {
  const { navigate } = useNavigation();
  const [isLoading, setIsLoading] = useState(false);

  const tryLogout = useCallback(() => {
    Alert.alert(
      'Sign out',
      'Are you sure?',
      [
        {
          text: 'Yes, Sign out',
          onPress: () => {
            logout();
          }
        },
        {
          text: 'Cancel',
          style: 'cancel'
        }
      ],
      {
        cancelable: true
      }
    );
  }, []);

  const logout = async () => {
    setIsLoading(true);
    try {
      await Client.logout();
    } catch (e) {
      Log.warn(TAG, 'logout', e);
    }
    Session.logout();
    navigate('Login');
    setIsLoading(false);
  };

  return (
    <View style={[styles.container, style]}>
      <SplitText title={'Session settings'} type={'Default'} />
      <View style={styles.content}>
        <View style={styles.info}>
          <Text style={styles.description}>
            This option allows you to safely exit the application.
          </Text>
        </View>
        <Button
          style={styles.button}
          isLoading={isLoading}
          title={'Sign out'}
          typeColor={'accentDark'}
          onPress={tryLogout}
        />
      </View>
    </View>
  );
}

export default SessionSettings;
