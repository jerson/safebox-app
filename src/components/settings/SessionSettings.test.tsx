import React from 'react';
import * as renderer from 'react-test-renderer';
import SessionSettings, {SessionSettingsProps} from './SessionSettings';

const props: SessionSettingsProps = {};
test('renders without crashing', () => {
  const snapshot = renderer.create(<SessionSettings {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<SessionSettings {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
