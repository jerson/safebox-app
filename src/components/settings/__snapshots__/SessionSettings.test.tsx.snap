// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fafdff",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#9066f5",
            "height": 1,
            "left": 0,
            "opacity": 0.5,
            "position": "absolute",
            "right": 0,
            "top": 12,
          },
          Object {
            "backgroundColor": "#c2d0e7",
          },
          false,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#4531cd",
            "borderRadius": 10,
            "paddingBottom": 4,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 6,
          },
          Object {
            "backgroundColor": "#6781a6",
          },
          false,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Nunito",
              "fontSize": undefined,
              "fontStyle": "normal",
              "fontWeight": "400",
            },
            Object {},
            Object {
              "color": "#FFFFFF",
              "fontSize": 13,
              "lineHeight": 14,
              "minWidth": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Session settings
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingRight": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Nunito",
              "fontSize": undefined,
              "fontStyle": "normal",
              "fontWeight": "400",
            },
            Object {},
            Object {
              "color": "#65748b",
              "fontSize": 12,
            },
          ]
        }
      >
        This option allows you to safely exit the application.
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#6781a6",
          "borderColor": "#6781a6",
          "borderRadius": 10,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 0,
          "minWidth": 120,
          "opacity": 1,
          "padding": 12,
          "zIndex": 1,
        }
      }
      testID="button_touchable"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Nunito",
              "fontSize": undefined,
              "fontStyle": "normal",
              "fontWeight": "400",
            },
            Object {},
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
              },
              undefined,
            ],
          ]
        }
      >
        Sign out
      </Text>
    </View>
  </View>
</View>
`;
